### ISSUE TITLE
Updating HelmRelease `spec.chart.ref` doesn't refetch new Git commits

### ISSUE THREAD

**Issue by ghost** [fluxcd/helm-operator#458]
**Describe the bug**

When updating the `spec.chart.ref` field in the `HelmRelease` to a recent commit, the Helm Operator fails to sync the chart. Only after the `--git-poll-interval` or forcing a sync by POSTing to `/api/v1/sync-git` does the release succeed.

**To Reproduce**

Steps to reproduce the behaviour:
1. Deploy the Helm Operator with a long `--git-poll-interval` to prevent it from syncing.
2. Apply a `HelmRelease` using a Git repository as the chart source.
3. Push a new commit to the Git repository hosting the chart.
4. Edit `spec.chart.ref` of the `HelmRelease` to the new commit SHA.
5. The `HelmRelease` should fail to update with a `FailedReleaseSync` warning below.

```
Warning  FailedReleaseSync  8m32s  helm-operator  synchronization of release 'my-service' in namespace 'my-namespace' failed: failed to prepare chart for release: chart unavailable: fatal: bad object cf2d9513396dcc96df7ad03c44c76776c66eb2ff, full output:
 fatal: bad object cf2d9513396dcc96df7ad03c44c76776c66eb2ff
```

**To fix the problem:**
Force sync Git with the following, or wait for the operator to poll Git again. I exec'ed into the Helm Operator pod and found that the repo in `/tmp/flux-gitclonexxxxxx` didn't have the latest commit until a sync.
```
kubectl port-forward -n flux deployment/helm-operator 3030:3030
curl -XPOST http://localhost:3030/api/v1/sync-git
```

I had a brief look through the code in this repo and think this would be the trace of function calls. I was looking at commit a3beeafe76a72aa082c5d19ec1a9ca6db1976e39.

```
pkg/chartsync/git.go:258
~/go/pkg/mod/github.com/fluxcd/flux@v1.18.0/pkg/git/repo.go:220
~/go/pkg/mod/github.com/fluxcd/flux@v1.18.0/pkg/git/operations.go:247
```

**Expected behavior**

When the `spec.chart.ref` is changed and the updated ref value is not found in the Git repo, fetch new commits before throwing an error.

**Additional context**

- Helm Operator version: 1.1.0
- Kubernetes version: v1.15.10
- Git provider: GitLab.com
- Helm repository provider: GitLab.com

**Comment by kingdonb** [comment]
Sorry if your issue remains unresolved. The Helm Operator is in maintenance mode, we recommend everybody upgrades to Flux v2 and Helm Controller.

A new release of Helm Operator is out this week, 1.4.4.

We will continue to support Helm Operator in maintenance mode for an indefinite period of time, and eventually archive this repository.

Please be aware that Flux v2 has a vibrant and active developer community who are actively working through minor releases and delivering new features on the way to General Availability for Flux v2.

In the mean time, this repo will still be monitored, but support is basically limited to migration issues only. I will have to close many issues today without reading them all in detail because of time constraints. If your issue is very important, you are welcome to reopen it, but due to staleness of all issues at this point a new report is more likely to be in order. Please open another issue if you have unresolved problems that prevent your migration in the appropriate Flux v2 repo.

Helm Operator releases will continue as possible for a limited time, as a courtesy for those who still cannot migrate yet, but these are strongly not recommended for ongoing production use as our strict adherence to semver backward compatibility guarantees limit many dependencies and we can only upgrade them so far without breaking compatibility. So there are likely known CVEs that cannot be resolved.

We recommend upgrading to Flux v2 which is actively maintained ASAP.

I am going to go ahead and close every issue at once today,
Thanks for participating in Helm Operator and Flux! 💚 💙
